### DISCUSSION TITLE
Help with testing in Livewire

### DISCUSSION THREAD

**Discussion by carlosoberto88** [livewire/livewire#5398] [1. Help]
Hello devs, 

I know this is probably a stupid question but I need help with some testing in a Livewire Application.

I have a form with multiple selects all related with each other, example: I have a country select, an state select and a city select. When the user selects a country I have a function called updatedCountrySelect which updates the state select, I really hope I'm making myself clear.

The selects are working in my form but I don't know how to write the test to run some automated test to validate that all this functions work correctly. I would love to be able to write a test that would take a random country and run all the scripts to update the subsequent selects and check there's no error. 

Hope somebody can help me.

**Comment by satoved** [comment]
Have you tried `->set('countrySelected', 'Germany')`?

**Comment by joshhanley** [comment]
@carlosoberto88 I would try and write the test yourself first and then share what you tried here, as it's easier for people to assist if they can see your code (if something doesn't quite work as expected). Have a look at the testing page in the docs, there is a lot of helpers, such as the one suggested above by @satoved. https://laravel-livewire.com/docs/2.x/testing
